In [2]:
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import plotly.express as px
import plotly.graph_objects as go

In [3]:
df = pd.read_csv("data.csv", index_col=0, delimiter=";")
df

Pohlavi  EQ  SQ  AQ
#                       
1          1  30  46  20
11         1  53  45   8
27         0  50  14  13
33         1  51  50  17
36         0  55  31  13
..       ...  ..  ..  ..
958        0  43  28  17
960        0  32  13  17
967        0  43  35  11
968        0  37  47  15
988        1  41  50  12

[120 rows x 4 columns]

In [4]:
df_women = df.query("Pohlavi == 0")
df_men = df.query("Pohlavi == 1")

In [5]:
df_men.describe()

Pohlavi         EQ         SQ         AQ
count     26.0  26.000000  26.000000  26.000000
mean       1.0  33.576923  38.192308  20.423077
std        0.0  10.911180   8.971150   6.694314
min        1.0  16.000000  20.000000   8.000000
25%        1.0  27.250000  32.000000  15.250000
50%        1.0  32.000000  38.000000  20.500000
75%        1.0  40.750000  44.750000  25.000000
max        1.0  54.000000  52.000000  33.000000

In [6]:
df_women.describe()

Pohlavi         EQ         SQ         AQ
count     94.0  94.000000  94.000000  94.000000
mean       0.0  39.074468  24.372340  16.936170
std        0.0  10.050667   9.464119   6.960423
min        0.0  13.000000   8.000000   5.000000
25%        0.0  33.000000  17.000000  12.000000
50%        0.0  40.000000  22.000000  16.000000
75%        0.0  45.750000  31.000000  21.000000
max        0.0  62.000000  47.000000  35.000000

Statistical tests scipy: https://docs.scipy.org/doc/scipy/reference/stats.html#statistical-tests


# Úkol 7: Welchův test
Britský psycholog Simon Baron-Cohen popsal dva osobnostní rysy, které pojmenoval Empatizace (schopnost a sklon rozumět druhým a zajímat se o jejich prožitky) a Systemizace (schopnost a sklon rozumět neživým, kauzálním, technickým soustavám, jako jsou stroje, rovnice či různé klasifikace). Výsledky výzkumů opakovaně ukazují, že v těchto rysech existují genderové rozdíly, které se mimo jiné projevují vyšší mírou projevů subklinického autismu u mužů.

Na souboru 120 respondentů platnost těchto výsledků ověřte. Pro každého respondenta máme k dispozici kromě údaje o jeho pohlaví, jeho skóry v dotazníku Systemizace (SQ), Empatizace (EQ) a subklinického autismu (AQ).

Předpokládejme, že všechny sledované veličiny mají přibližně normální rozdělení, shoda rozptylů mezi skupinami však není garantována. Není-li řečeno jinak, uvádějte hodnoty testové statistiky t bez znaménka.

=====================================================

Průměrné výsledky inventáře EQ jsou rozdílné u mužů a žen.

In [8]:
a = df_women["EQ"].values
b = df_men["EQ"].values

stats.ttest_ind(a, b, equal_var=False, alternative="two-sided")

Ttest_indResult(statistic=2.3120916831683576, pvalue=0.02635482364747578)

Průměrné výsledky inventáře SQ jsou rozdílné u mužů a žen..

In [9]:
a = df_women["SQ"].values
b = df_men["SQ"].values

stats.ttest_ind(a, b, equal_var=False, alternative="two-sided")

Ttest_indResult(statistic=-6.868629049113808, pvalue=2.322515905292222e-08)

Průměrné výsledky inventáře AQ jsou rozdílné u mužů a žen.

In [15]:
a = df_women["AQ"].values
b = df_men["AQ"].values

aq_test = stats.ttest_ind(a, b, equal_var=False, alternative="two-sided")
aq_test

Ttest_indResult(statistic=-2.3303034159688205, pvalue=0.02476622098787011)

Pro výše uvedenou statistiku uveďte počet stupňů volnosti.

In [14]:
n1 = len(a)
n2 = len(b)

v1 = np.var(a, ddof=1)
v2 = np.var(b, ddof=1)

vn1 = v1 / n1
vn2 = v2 / n2

df = (vn1 + vn2)**2 / (vn1**2 / (n1 - 1) + vn2**2 / (n2 - 1))
df

41.19631524609479

O kolik směrodatných odchylek se liší muži a ženy v AQ. Spočítejte Cohenovo d (viz definice ve skriptech). Napište jeho hodnotu včetně znaménka - plusem označte vyšší skór mužů, mínusem žen.

In [19]:
cohen_d = aq_test.statistic * np.sqrt(1/n1 + 1/n2)
cohen_d

-0.5163598833719936

In [22]:
a = df_men["AQ"].values
b = df_women["AQ"].values

m1 = np.mean(a)
m2 = np.mean(b)

n1 = len(a)
n2 = len(b)

v1 = np.var(a, ddof=1)
v2 = np.var(b, ddof=1)

var = (v1 * (n1 - 1) + v2 * (n2 - 1)) / (n1 + n2 - 2)
std = np.sqrt(var)

cohen_d = (m1 - m2) / std
cohen_d, std

(0.5049901293333348, 6.904900725315606)

The second one is correct